addons/product/models/product_template.py

```diff
@@ -824,16 +831,24 @@ def _get_variant_for_combination(self, combination):
 
     @api.multi
     def _get_first_possible_combination(self, parent_combination=None, necessary_values=None):
+        """See `_get_nth_possible_combination` with n=1"""
+        return self._get_nth_possible_combination(parent_combination, necessary_values, n=1)
+
+    @api.multi
+    def _get_nth_possible_combination(self, parent_combination=None, necessary_values=None, n=1):
```

Comment:
don't you have a function with an iterator that return each combinaisons?
If not, probably a good thing to do, and just iters 2 times to check if more than one.

Reply:
Done, 1st commit updated by using generator instead of get_nth. This doesn't have a significant performance impact, but that code is more understandable this way (and not more nonlocal).